internal/k8s/builder/testdata/tls-cert.deployment.golden.yaml

```diff
@@ -39,27 +39,31 @@ spec:
               topologyKey: kubernetes.io/hostname
             weight: 1
       containers:
-      - command:
-        - /bin/sh
-        - -ec
-        - |2-
-
-          export CONSUL_CACERT=/consul/tls/ca.pem
-          cat <<EOF >/consul/tls/ca.pem
-          CONSUL_CA_MOCKED
-          EOF
```

Comment:
This looks a little different from other tests, should any of this be reflected in the new CA cert logic?

Reply:
The `CONSUL_CACERT` envvar is now added to the pod spec below in this case, and the cert content ("CONSUL_CA_MOCKED" here) would now be deployed into a secret that the volume mount below references.

I did refine the code a bit just now to only include the envvar and create the secret when the CA is actually required, so now it's clearer that the resulting deployment for this test case which _does_ require the CA is different from the others which don't require the CA.